modules/polygon/src/polygon.js

```diff
@@ -1,20 +1,36 @@
-import {equals} from '@math.gl/core';
+/* eslint-disable no-undef */
+/* eslint-disable no-console */
+
+import {isArray} from '@math.gl/core';
+
+import {
+  getPolygonSignedArea,
+  getPolygonSignedAreaFlat,
+  forEachSegmentInPolygon,
+  forEachSegmentInPolygonFlat,
+  ensurePolygonWindingDirection,
+  ensurePolygonWindingDirectionFlat
+} from './polygon-utilities';
 
 export default class Polygon {
-  constructor(points) {
+  constructor(points, options = {}) {
     this.points = points;
-    this.isClosed = equals(this.points[this.points.length - 1], this.points[0]);
+    this.isFlatArray = !isArray(points[0]);
+
+    if (this.isFlatArray) {
+      this.start = options.start || 0;
+      this.end = options.end || points.length;
```

Comment:
Nit: in loaders' [binary format](https://github.com/visgl/loaders.gl/blob/master/modules/gis/docs/api-reference/geojson-to-binary.md#outputs) we count each coordinate as 1, not as 2 or 3. `points.length` implicitly counts each coordinate as the size of that coordinate.

Reply:
Also, instead of dealing with `start` and `end` directly, why not take a view of the input? Is the flat array input allowed to not be a `TypedArray`? Wouldn't seem to be a problem to accept either an array of arrays or a flat `TypedArray`.

Reply:
> 

> Also, instead of dealing with start and end directly, why not take a view of the input? Is the flat array input allowed to not be a TypedArray? Wouldn't seem to be a problem to accept either an array of arrays or a flat TypedArray.

Depends on the number of views. Views are not free unfortunately. Each view is a small object that needs to be garbage collected. E.g. taking 100K views for a long binary array with 100K polygons does have a GC cost. So if we worry enough about not having a small object for params, then we probably don't want to rely on views.

Reply:
Thanks, I think I mistakenly thought that views were "small enough to be free".